File: infra/Dockerfile.py
FROM python:3.11-slim

# System basics (optional but handy)
RUN apt-get update -qq && apt-get install -y --no-install-recommends \
    curl && rm -rf /var/lib/apt/lists/*

# Python deps (FastAPI, server, and HTTP client)
RUN pip install --no-cache-dir fastapi uvicorn httpx pydantic

WORKDIR /app
COPY . /app

File: infra/Dockerfile.r
FROM rocker/r-base:4.4.1

RUN apt-get update -qq && apt-get install -y --no-install-recommends \
libssl-dev libcurl4-openssl-dev libxml2-dev && \
rm -rf /var/lib/apt/lists/*
  
  RUN R -e "install.packages(c('plumber','jsonlite','httr'), repos='https://cloud.r-project.org')"

WORKDIR /app
COPY . /app
